hivemind/compression/floating.py

```diff
@@ -12,22 +12,28 @@ class Float16Compression(CompressionBase):
     FP16_MIN, FP16_MAX = torch.finfo(torch.float16).min, torch.finfo(torch.float16).max
 
     def compress(self, tensor: torch.Tensor, info: CompressionInfo, allow_inplace: bool = False) -> runtime_pb2.Tensor:
+        assert torch.is_floating_point(tensor) and tensor.dtype != torch.bfloat16
```

Comment:
Is there a reason why we should fail with an error in case of bf16 inputs? It is indeed not sensible, but  if the user wants to do so, it's probably better to issue a warning instead of flat out refusing to pass that through quantization

Reply:
added ValueError with a more user-legible reason